### PR TITLE
csi: Add nfs snap & clone ci

### LINKED ISSUE #10895: csi: add NFS snap, clone e2e
Please have a TODO to add E2E test cases for this

Depends on: #10518

_Originally posted by @Madhu-1 in https://github.com/rook/rook/pull/10892#pullrequestreview-1097435758_

### PR DESCRIPTION
**Pull request by Rakshith-R**
**Description of your changes:**
- ci: add nfs sna & restore e2e testcase
- ci: add nfs clone e2e testcase

**Which issue is resolved by this Pull Request:**
Resolves #10895 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: tests/integration/ceph_base_nfs_test.go [hunk 1]
```diff
@@ -50,9 +54,136 @@ func runNFSFileE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s *sui
 
 		cleanupFilesystemConsumer(helper, k8sh, s, settings.Namespace, filePodName)
 		assert.NoError(s.T(), err)
+
+		nfsSnapshotTest(helper, k8sh, s, storageClassName, settings.Namespace)
```
[the remark this answers is not in the document]
Comment (Rakshith-R) [reply tests/integration/ceph_base_nfs_test.go]: simplified it. PTAL

File: tests/integration/ceph_base_file_test.go [hunk 2]
```diff
@@ -233,9 +238,13 @@ func fileSystemCSISnapshotTest(helper *clients.TestClient, k8sh *utils.K8sHelper
 	assert.True(s.T(), k8sh.WaitUntilPVCIsDeleted(defaultNamespace, pvcName))
 
 	logger.Infof("delete snapshotclass")
-
-	err = helper.FSClient.DeleteSnapshotClass(snapshotClassName, snapshotDeletePolicy, namespace)
-	require.NoError(s.T(), err)
+	if !testNFS {
+		err = helper.FSClient.DeleteSnapshotClass(snapshotClassName, snapshotDeletePolicy, namespace)
+		require.NoError(s.T(), err)
```
Comment (Madhu-1) [inline tests/integration/ceph_base_file_test.go:243]: require.NoError can be outside of if else check

File: tests/integration/ceph_base_file_test.go [hunk 3]
```diff
@@ -142,8 +142,13 @@ func fileSystemCSISnapshotTest(helper *clients.TestClient, k8sh *utils.K8sHelper
 	snapshotDeletePolicy := "Delete"
 	snapshotClassName := "snapshot-testing"
 	logger.Infof("create snapshotclass")
-	err = helper.FSClient.CreateSnapshotClass(snapshotClassName, snapshotDeletePolicy, namespace)
-	require.NoError(s.T(), err)
+	if !testNFS {
+		err = helper.FSClient.CreateSnapshotClass(snapshotClassName, snapshotDeletePolicy, namespace)
+		require.NoError(s.T(), err)
```
Comment (Madhu-1) [inline tests/integration/ceph_base_file_test.go:147]: require can be outside of if else check

**Review by Madhu-1 — APPROVED** [review]

File: tests/integration/ceph_base_file_test.go (hunk 3 above)
Comment (Rakshith-R) [reply tests/integration/ceph_base_file_test.go]: done :+1:

File: tests/integration/ceph_base_file_test.go (hunk 2 above)
Comment (Rakshith-R) [reply tests/integration/ceph_base_file_test.go]: done :+1:

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**